cleanlab/dataset.py

```diff
@@ -0,0 +1,416 @@
+        print(
+            f" * Overall, about {1 - joint_trace:.0%} ({num_issues:,} of the {num_examples:,}) "
+            f"labels in your dataset have issues.\n"
+            f" ** The overall label health score for this dataset is: {joint_trace:.2f}."
+        )
+    return joint_trace
+
+
+def health_summary(
+    labels=None,
+    pred_probs=None,
+    *,
+    asymmetric=False,
+    class_names=None,
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Prints a healthy summary of your datasets including results for powerful statistics like:
+    * the classes with the most and least label issues
+    * classes that overlap and could potentially be merged
+    * overall data label quality health score statistics for your dataset
+
+    Parameters
+    ----------
+    For parameter info, see the docstring of `dataset.find_overlapping_classes`
+
+    Returns
+    -------
+    health_score : float
+        A score between 0 and 1 where 1 implies the dataset has all estimated perfect labels.
+        A score of 0.5 implies that, on average, half of the dataset's label have estimated issues.
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+
+    print(
+        "*" * 56
+        + "\n"
+        + "Generating a Cleanlab Dataset Health Summary\n"
+        + f" for your dataset with {num_examples:,} examples and {len(joint):,} classes.\n"
+        + "Note, Cleanlab is not a medical doctor... yet.\n"
+        + "*" * 56
+        + "\n",
+    )
+
+    df_class_label_quality = rank_classes_by_label_quality(
+        labels=labels,
+        pred_probs=pred_probs,
+        class_names=class_names,
+        num_examples=num_examples,
+        joint=joint,
+        confident_joint=confident_joint,
+        multi_label=multi_label,
+    )
+    print("Overall Class Quality and Noise across your dataset (below)")
+    print("-" * 60, "\n", flush=True)
+    print(df_class_label_quality)
+
+    df_class_label_quality = find_overlapping_classes(
+        labels=labels,
+        pred_probs=pred_probs,
+        asymmetric=asymmetric,
+        class_names=class_names,
+        num_examples=num_examples,
+        joint=joint,
+        confident_joint=confident_joint,
+        multi_label=multi_label,
+    )
+    print(
+        "\nClass Overlap. In some cases, you may want to merge classes in the top rows (below)"
+        + "\n"
+        + "-" * 83
+        + "\n",
+        flush=True,
+    )
+    print(df_class_label_quality)
+    print()
+    health_score = overall_label_health_score(labels=labels, pred_probs=pred_probs, verbose=True)
+    print("\nGenerated with <3 from Cleanlab.")
+    return health_score
```

Comment:
`health_summary()` seems to already compute the results of every other function in this module, but they're being tossed. Any way to return these to user as well?

This speaks to the need for future `Dataset` class to store intermediate results. 

For now, this method could have boolean flag argument: `return_value :  List[str]` with default value: `["health_score", "class_label_quality", "class_overlap"]` (and another option for this list could be `"joint"`). Then 
based on `return_value`, this method returns a dict whose keys = the values in `return_value` and values = the corresponding scores / dataframes / arrays.